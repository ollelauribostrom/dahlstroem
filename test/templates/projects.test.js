import React from 'react';
import { shallow } from 'enzyme';

import ProjectsTemplate from '../../src/templates/projects';

const mockData = {
  allDatoCmsWork: {
    edges: [
      {
        node: {
          title: 'Test project',
          slug: '/test',
          images: [
            {
              fluid: {
                aspectRatio: 1.4678899082568808,
                sizes: '(max-width: 1800px) 100vw, 1800px',
                src: 'someurl1',
                srcSet: 'someurl1',
              },
            },
            {
              fluid: {
                aspectRatio: 1.4678899082568808,
                sizes: '(max-width: 1800px) 100vw, 1800px',
                src: 'someurl2',
                srcSet: 'someurl2',
              },
            },
          ],
        },
      },
    ],
  },
};

describe('Tests for Projects template', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ProjectsTemplate
        location={{ pathname: '/test' }}
        pageContext={{ seo: { title: 'test', keywords: ['a', 'b'] } }}
        data={mockData}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly without any projects', () => {
    const wrapper = shallow(
      <ProjectsTemplate
        location={{ pathname: '/test' }}
        pageContext={{ seo: { title: 'test', keywords: ['a', 'b'] } }}
        data={{
          noContentImage: {
            childImageSharp: {
              fluid: {
                aspectRatio: 1.4678899082568808,
                sizes: '(max-width: 1800px) 100vw, 1800px',
                src: 'someurl2',
                srcSet: 'someurl2',
              },
            },
          },
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
