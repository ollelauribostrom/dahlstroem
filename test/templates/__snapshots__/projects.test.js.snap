// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Projects template renders correctly 1`] = `
<Layout
  color={
    Object {
      "hex": "#000",
    }
  }
  menuBackground={
    Object {
      "hex": "#000",
    }
  }
  menuColor={
    Object {
      "hex": "#fff",
    }
  }
  path="/test"
>
  <SEO
    keywords={
      Array [
        "a",
        "b",
      ]
    }
    lang="en"
    meta={Array []}
    title="test"
  />
  <div
    className="projects"
  >
    <ProjectThumbnail
      key="/test"
      project={
        Object {
          "images": Array [
            Object {
              "fluid": Object {
                "aspectRatio": 1.4678899082568808,
                "sizes": "(max-width: 1800px) 100vw, 1800px",
                "src": "someurl1",
                "srcSet": "someurl1",
              },
            },
            Object {
              "fluid": Object {
                "aspectRatio": 1.4678899082568808,
                "sizes": "(max-width: 1800px) 100vw, 1800px",
                "src": "someurl2",
                "srcSet": "someurl2",
              },
            },
          ],
          "slug": "/test",
          "title": "Test project",
        }
      }
    />
  </div>
</Layout>
`;

exports[`Tests for Projects template renders correctly without any projects 1`] = `
<Layout
  color={
    Object {
      "hex": "#000",
    }
  }
  menuBackground={
    Object {
      "hex": "#000",
    }
  }
  menuColor={
    Object {
      "hex": "#fff",
    }
  }
  path="/test"
>
  <SEO
    keywords={
      Array [
        "a",
        "b",
      ]
    }
    lang="en"
    meta={Array []}
    title="test"
  />
  <div
    className="projects"
  >
    <Image
      Tag="div"
      alt=""
      critical={false}
      fadeIn={true}
      fluid={
        Object {
          "aspectRatio": 1.4678899082568808,
          "sizes": "(max-width: 1800px) 100vw, 1800px",
          "src": "someurl2",
          "srcSet": "someurl2",
        }
      }
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</Layout>
`;
